File: PYTHON/Komputer.py
from Cpu import Cpu
from Ram import Ram
from Harddrive import Harddrive
from Motherboard import Motherboard
from PowerSupply import PowerSupply
from Gpu import Gpu
from SmartPowerSupply import SmartPowerSupply



class Komputer:
    def __init__(self, nama="", cpu=None, ram_list=None, gpu_list=None, harddrive=None, motherboard=None, psu=None):
        self.nama = nama
        self.cpu = cpu
        self.ram_list = ram_list if ram_list is not None else []
        self.gpu_list = gpu_list if gpu_list is not None else []
        self.harddrive = harddrive
        self.motherboard = motherboard
        self.psu = psu

    def set_nama(self, nama):
        self.nama = nama

    def set_cpu(self, cpu):
        self.cpu = cpu

    def set_ram(self, ram_list):
        self.ram_list = ram_list

    def set_gpu(self, gpu_list):
        self.gpu_list = gpu_list

    def set_harddrive(self, harddrive):
        self.harddrive = harddrive

    def set_motherboard(self, motherboard):
        self.motherboard = motherboard

    def set_psu(self, psu):
        self.psu = psu

    def add_ram(self, ram):
        self.ram_list.append(ram)

    def add_gpu(self, gpu):
        self.gpu_list.append(gpu)

    def get_nama(self):
        return self.nama

    def get_cpu(self):
        return self.cpu

    def get_ram_list(self):
        return self.ram_list

    def get_gpu_list(self):
        return self.gpu_list

    def get_harddrive(self):
        return self.harddrive

    def get_motherboard(self):
        return self.motherboard

    def get_psu(self):
        return self.psu

File: CPP/Komputer.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"
#include "Motherboard.cpp"
#include "PowerSupply.cpp"
#include "Gpu.cpp"



using namespace std;


class Komputer
{
private:
    string nama;
    Cpu cpu;
    vector<Ram> ramList;
    Harddrive harddrive;
    Motherboard motherboard;
    PowerSupply psu;
    vector<Gpu> gpuList;

public:
    Komputer()
    {
    }


    Komputer(string nama, Cpu cpu, vector<Ram> ramList, vector<Gpu> gpuList, Harddrive harddrive, Motherboard motherboard, PowerSupply psu)
    {
        this->nama = nama;
        this->cpu = cpu;
        this->ramList = ramList;
        this->gpuList = gpuList;
        this->harddrive = harddrive;
        this->motherboard = motherboard;
        this->psu = psu;
    }

    void setNama(string nama)
    {
        this->nama = nama;
    }

    void setCpu(Cpu cpu)
    {
        this->cpu = cpu;
    }


    void setRam(vector<Ram> ramList)
    {
        this->ramList = ramList;
    }    

    void setGpu(vector<Gpu> gpuList)
    {
        this->gpuList = gpuList;
    }
   
    void setHarddrive(Harddrive harddrive)
    {
        this->harddrive = harddrive;
    }

    void setMotherboard(Motherboard motherboard)
    {
        this->motherboard = motherboard;
    }

    void setPsu(PowerSupply psu)
    {
        this->psu = psu;
    }

    void addRam(Ram ram)
    {
        this->ramList.push_back(ram);
    }

    void addGpu(Gpu gpu)
    {
        this->gpuList.push_back(gpu);
    }


    string getNama()
    {
        return this->nama;
    }


    Cpu getCpu()
    {
        return this->cpu;
    }


    vector<Ram> getRamList() // return vector
    {
        return this->ramList;
    }

    vector<Gpu> getGpuList() // return vector
    {
        return this->gpuList;
    }


    Harddrive getHarddrive()
    {
        return this->harddrive;
    }

    Motherboard getMotherboard()
    {
        return this->motherboard;
    }

    PowerSupply getPsu()
    {
        return this->psu;
    }

    ~Komputer()
    {
    }


};
